src/api/java/baritone/api/utils/ExampleBaritoneControl.java

```diff
@@ -662,10 +719,43 @@ public boolean isInGoal(int x, int y, int z) {
         }
         if (msg.equals("damn")) {
             logDirect("daniel");
+            return true;
+        }
+        if (msg.startsWith("add")) {
+            String newCommands = msg.substring(3).trim();
+            commandsToBeExecuted = commandsToBeExecutedOld + newCommands;
+            return true;
+        }
+        if (msg.equals("next")) {
+            commandsToBeExecuted = commandsToBeExecutedOld;
+            nextCommand();
+            return true;
+        }
+        if (msg.equals("commandlist")) {
+            commandsToBeExecuted = commandsToBeExecutedOld;
+            logDirect(commandsToBeExecuted);
+            return true;
         }
         return false;
     }
 
+    public static void nextCommand() { // execute nex command in commandsToBeExecuted
+        String[] commandsList = commandsToBeExecuted.split(COMMAND_PREFIX);
+        if (commandsList.length > 1) {
+            String msg0 = commandsList[1];
+            StringBuilder sb = new StringBuilder();
+            for (int i = 2; i < commandsList.length; i++) {
+                sb.append(COMMAND_PREFIX);
+                sb.append(commandsList[i]);
+            }
+            commandsToBeExecuted = sb.toString();
+            thisExampleBaritoneControl.logDirectTextComponentString(new TextComponentString(String.format("%s Executing: %s" + COMMAND_PREFIX + msg0, TextFormatting.YELLOW, TextFormatting.WHITE)));
```

Comment:
does not look like `logDirectTextComponentString` is required at all lol. remove `logDirectTextComponentString` and replace it with `logDirect` it should work perfectly fine.